test/onnx/test_fx_to_onnx_with_onnxruntime.py

```diff
@@ -1067,23 +1067,48 @@ def _test_fake_tensor_mode_exporter(
                 # Generate random inputs.
                 args = create_args()
                 kwargs = create_kwargs()
+
                 # Original outputs.
-                # model_with_state_dict=real_model is used to create non-fake weights
-                ref_outputs = onnx_program.adapt_torch_outputs_to_onnx(
-                    real_model(*args, **kwargs), model_with_state_dict=real_model
-                )
+                if (
+                    model_type
+                    == pytorch_test_common.TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAM
+                ):
+                    ref_outputs = onnx_program.adapt_torch_outputs_to_onnx(
+                        real_model(*args, **kwargs), model=real_model
+                    )
+                elif model_type == pytorch_test_common.TorchModelType.TORCH_NN_MODULE:
+                    ref_outputs = onnx_program.adapt_torch_outputs_to_onnx(
+                        real_model(*args, **kwargs)
+                    )
                 # ORT outputs.
-                # model_with_state_dict=real_model is used to create non-fake weights
-                args_not_none = onnx_program.adapt_torch_inputs_to_onnx(
-                    *args, model_with_state_dict=real_model, **kwargs
-                )
+                # exported_program=real_model is used to create non-fake weights
+                if (
+                    model_type
+                    == pytorch_test_common.TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAM
+                ):
+                    input_names, _ = onnx_test_common.get_ort_input_names_and_session(
+                        tmp_onnx_file.name
+                    )
+                    args_not_none = onnx_program.adapt_torch_inputs_to_onnx(
+                        *args,
+                        exported_program=real_model,
```

Comment:
I see the workaround you were talking about in the meeting.

IIUC, in order for fake exported model to have non-persistent buffers, we need to concretely torch.export.export the model, and stash the buffers back?

Reply:
How about we hijack the initialization of non-persistent buffer and record a concrete tensor under our export-purpose fake mode context manager when the model was being initialized? And non-persistent buffers initialized at runtime should get traced so we should have everything covered?

Reply:
@thiagocrepaldi What do you think? Serialize non-persistent buffers in fake mode context manager might be doable.